### ISSUE TITLE
Show an error if auth type is incorrect

### ISSUE THREAD

**Issue by preslavmihaylov** [preslavmihaylov/todocheck#71]
Here's the error I get when the provided auth type in `.todocheck.yaml` is incorrect:
![image](https://user-images.githubusercontent.com/7642330/95826513-174bb080-0d3b-11eb-9ddd-04ebf85f56f1.png)

This happens because there is a check for it in one of the components using the config. It would be more appropriate to validate against this in the `config.Validate` function instead and show an error message to the user in a normal fashion, rather than relying on a panic, which happens afterwards.

**Comment by akk5597** [comment]
Would the solution be to call `authmanager.AcquireToken` in `config.Validate`, and replace the `panic` with an `error` saying that the auth token type is unknown?

**Comment by preslavmihaylov** [comment]
I think a good approach would be to make a function like this (taken from [here](https://www.ribice.ba/golang-enums/):
```
func (lt LeaveType) IsValid() error {
    switch lt {
    case AnnualLeave, Sick, BankHoliday, Other:
        return nil
    }
    return errors.New("Invalid leave type")
}
```

This function should be in the same file as the enum declarations to avoid someone creating a new enum in the future without upgrading this func.

Using `AcquireToken` to validate is not a good idea, IMO, because it is not meant to be used for validation, but rather, for acquiring a token as the name states.

**Comment by zen37** [comment]
> It would be more appropriate to validate against this in the config.Validate function instead 

Where is the config.Validate function? I could not find it.

**Comment by preslavmihaylov** [comment]
https://github.com/preslavmihaylov/todocheck/blob/ce4a2420e46ddfc5795358363e05399d874a9b4f/config/config.go#L145

**Comment by zen37** [comment]
sorry, I didn't know of this convention, fileName.functionName and it is logical.

**Comment by preslavmihaylov** [comment]
Closed by #84
